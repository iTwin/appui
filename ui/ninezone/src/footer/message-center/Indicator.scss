/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "../../_geometry";
@import "../_variables";

.nz-footer-messageCenter-indicator {
  height: 100%;
  display: flex;
  align-items: center;
  padding-left: $nz-footer-indicator-horizontal-padding;
  padding-right: $nz-footer-indicator-horizontal-padding;

  > .nz-label {
    margin-right: $space-between-text-and-icon;
    display: none;
    font-size: $text-font-size;

    &.nz-is-visible {
      display: block;
    }
  }

  > .nz-balloon-container {
    height: 100%;
    position: relative;
    display: grid;

    > .nz-balloon {
      background-color: $text-font-color;
      text-align: center;
      cursor: pointer;
      min-width: $min-balloon-width;
      min-height: $min-balloon-height;
      border-radius: 50%;
      padding: $balloon-padding;
      filter: $icon-shadow;
      position: relative;
      align-self: center;

      > .nz-arrow {
        position: absolute;
        width: 0;
        bottom: -3px;
        left: 0;
        display: block;
        border-width: 0;
        border-bottom-width: $balloon-arrow-height;
        border-left-width: $balloon-arrow-width;
        border-style: solid;
        border-color: transparent $text-font-color;
        transform: rotate(25deg);
      }

      > .nz-content {
        position: relative;
        color: #FFFFFF;
        line-height: $min-balloon-height;
        font-size: 11px;
      }
    }

    > .nz-dialog {
      height: 100%;
      width: 100%;
      position: absolute;
      top: 0;
      left: 50%;
    }
  }
}
