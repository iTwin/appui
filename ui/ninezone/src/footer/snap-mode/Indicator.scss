/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "@bentley/ui-core/lib/colors";
@import "../_variables";
.nz-footer-snapMode-indicator {
  height: 100%;
  display: flex;
  align-items: center;
  position: relative;
  padding-left: $nz-footer-indicator-horizontal-padding;
  padding-right: $nz-footer-indicator-horizontal-padding;
  >.nz-indicator {
    position: relative;
    display: flex;
    align-items: center;
    >.nz-label {
      margin-right: $space-between-text-and-icon;
      display: none;
      font-size: $text-font-size;
      &.nz-is-visible {
        display: block;
      }
    }
  }
  >.nz-dialog {
    height: 100%;
    position: absolute;
    top: 0;
    left: 100%;
  }
}

.nz-footer-snapMode-indicator:hover {
  background-color: $primary-hover;
  cursor: pointer;
}