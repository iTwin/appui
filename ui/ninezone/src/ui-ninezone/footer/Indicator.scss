/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "@bentley/ui-core/lib/ui-core/style/themecolors";
@import "_variables";

.nz-footer-indicator {
  height: 100%;
  display: flex;
  align-items: center;
  color: $buic-text-color;
  font-size: $text-font-size;
  padding-top: $nz-footer-indicator-vertical-padding;
  padding-bottom: $nz-footer-indicator-vertical-padding;
  box-sizing: border-box;

  &.nz-footer-mode {
    padding-left: $nz-footer-indicator-horizontal-padding;
    padding-right: $nz-footer-indicator-horizontal-padding;
  }

  &:not(.nz-footer-mode) {
    padding-left: $nz-footer-widget-indicator-horizontal-padding;
    padding-right: $nz-footer-widget-indicator-horizontal-padding;
  }

  /* Medium devices (landscape tablets, 768px and up) */
  @media only screen and (max-width: 768px) {
    font-size: $text-font-size-medium;

    &.nz-footer-mode {
      padding-left: $nz-footer-indicator-horizontal-padding-medium;
      padding-right: $nz-footer-indicator-horizontal-padding-medium;
    }

    &:not(.nz-footer-mode) {
      padding-left: $nz-footer-widget-indicator-horizontal-padding-medium;
      padding-right: $nz-footer-widget-indicator-horizontal-padding-medium;
    }
  }

  /* small devices (phones, 600px and down) */
  @media only screen and (max-width: 600px) {
    font-size: $text-font-size-small;

    &.nz-footer-mode {
      padding-left: $nz-footer-indicator-horizontal-padding-small;
      padding-right: $nz-footer-indicator-horizontal-padding-small;
    }

    &:not(.nz-footer-mode) {
      padding-left: $nz-footer-widget-indicator-horizontal-padding-small;
      padding-right: $nz-footer-widget-indicator-horizontal-padding-small;
    }
  }
}
