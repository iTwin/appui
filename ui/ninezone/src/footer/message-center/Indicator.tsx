/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module MessageCenter */

import * as classnames from "classnames";
import * as React from "react";
import CommonProps, { NoChildrenProps } from "../../utilities/Props";
import "./Indicator.scss";

/** Properties of [[MessageCenterIndicator]] component. */
export interface MessageCenterIndicatorProps extends CommonProps, NoChildrenProps {
  /** Label of balloon icon. */
  balloonLabel?: string;
  /** Dialog that is opened when indicator is clicked. See [[MessageCenter]] */
  dialog?: React.ReactChild;
  /** Describes if the indicator label is visible. */
  isLabelVisible?: boolean;
  /** Indicator label. */
  label?: string;
  /** Function called when indicator is clicked. */
  onClick?: () => void;
}

/** One of [[Footer]] indicators. */
export default class MessageCenterIndicator extends React.Component<MessageCenterIndicatorProps> {
  public render() {
    const className = classnames(
      "nz-footer-messageCenter-indicator",
      this.props.className);

    const labelClassName = classnames(
      "nz-label",
      this.props.isLabelVisible && "nz-is-visible",
    );

    return (
      <div
        className={className}
        style={this.props.style}
      >
        <span className={labelClassName}>{this.props.label}</span>
        <div className="nz-balloon-container">
          <div className="nz-dialog">
            {this.props.dialog}
          </div>
          <div
            className="nz-balloon"
            onClick={this._handleOnIndicatorClick}
          >
            <div className="nz-arrow" />
            <div className="nz-content">
              {this.props.balloonLabel}
            </div>
          </div>
        </div>
      </div>
    );
  }

  private _handleOnIndicatorClick = () => {
    this.props.onClick && this.props.onClick();
  }
}
