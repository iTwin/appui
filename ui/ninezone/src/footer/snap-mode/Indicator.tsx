/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module SnapMode */

import * as classnames from "classnames";
import * as React from "react";
import CommonProps, { NoChildrenProps } from "../../utilities/Props";
import "./Indicator.scss";

/** Properties of [[SnapModeIndicator]] component. */
export interface SnapModeIndicatorProps extends CommonProps, NoChildrenProps {
  /** Dialog that is opened when indicator is clicked. See [[SnapModeDialog]] */
  dialog?: React.ReactChild;
  /** Indicator icon. I.e. [[SnapModeIcon]] */
  icon?: React.ReactNode;
  /** Describes if the label is visible. Pass true if the [[Footer]] is in footer mode. */
  isLabelVisible?: boolean;
  /** Indicator label. */
  label?: string;
  /** Function called when indicator is clicked. */
  onClick?: () => void;
}

/** One of [[Footer]] indicators. */
export default class SnapModeIndicator extends React.Component<SnapModeIndicatorProps> {
  public render() {
    const className = classnames(
      "nz-footer-snapMode-indicator",
      this.props.className);

    const labelClassName = classnames(
      "nz-label",
      this.props.isLabelVisible && "nz-is-visible",
    );

    return (
      <div
        className={className}
        style={this.props.style}
        onClick={this._handleOnIndicatorClick}
      >
        <div className="nz-indicator">
          <span className={labelClassName}>{this.props.label}</span>
          <div
            className="nz-icon"
          >
            {this.props.icon}
          </div>
        </div>
        <div className="nz-dialog">
          {this.props.dialog}
        </div>
      </div>
    );
  }

  private _handleOnIndicatorClick = () => {
    this.props.onClick && this.props.onClick();
  }
}
