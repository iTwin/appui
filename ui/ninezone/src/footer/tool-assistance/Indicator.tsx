/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module ToolAssistance */

import * as classnames from "classnames";
import * as React from "react";
import CommonProps, { NoChildrenProps } from "../../utilities/Props";
import "./Indicator.scss";

/** Properties of [[ToolAssistanceIndicator]] component. */
export interface ToolAssistanceIndicatorProps extends CommonProps, NoChildrenProps {
  /** Dialog that is opened when indicator is clicked. See [[ToolAssistanceDialog]] */
  dialog?: React.ReactNode;
  /** Indicator icons. */
  icons?: React.ReactNode;
  /** Describes if the step string is visible. */
  isStepStringVisible?: boolean;
  /** Function called when indicator is clicked. */
  onClick?: () => void;
  /** Step string. */
  stepString?: string;
}

/** One of [[Footer]] indicators. */
export default class ToolAssistanceIndicator extends React.Component<ToolAssistanceIndicatorProps> {
  public render() {
    const className = classnames(
      "nz-footer-toolAssistance-indicator",
      this.props.className);

    const stepStringClassName = classnames(
      "nz-step-string",
      this.props.isStepStringVisible && "nz-is-visible",
    );

    return (
      <div
        className={className}
        style={this.props.style}
      >
        <div
          className="nz-indicator"
          onClick={this._handleOnIndicatorClick}
        >
          <div className="nz-icons">
            {this.props.icons}
          </div>
          <span className={stepStringClassName}>{this.props.stepString}</span>
          <div className="nz-triangle" />
        </div>
        {this.props.dialog}
      </div>
    );
  }

  private _handleOnIndicatorClick = () => {
    this.props.onClick && this.props.onClick();
  }
}
