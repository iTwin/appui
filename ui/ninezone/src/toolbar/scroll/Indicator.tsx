/*---------------------------------------------------------------------------------------------
|  $Copyright: (c) 2018 Bentley Systems, Incorporated. All rights reserved. $
 *--------------------------------------------------------------------------------------------*/
/** @module Toolbar */

import * as classnames from "classnames";
import * as React from "react";

import CommonProps from "../../utilities/Props";
import Direction, { DirectionHelpers } from "../../utilities/Direction";
import "./Indicator.scss";
import Arrow from "./Arrow";

export interface IndicatorProps extends CommonProps {
  direction?: Direction;
  onScroll?: () => void;
}

export default class Indicator extends React.Component<IndicatorProps> {
  public render() {
    const className = classnames(
      "nz-toolbar-scroll-indicator",
      DirectionHelpers.getCssClassName(this.props.direction || Direction.Left),
      this.props.className);

    return (
      <div
        className={className}
        style={this.props.style}
      >
        <Arrow
          className={"nz-indicator"}
          onClick={this.props.onScroll}
          direction={this.props.direction}
        />
      </div>
    );
  }
}
