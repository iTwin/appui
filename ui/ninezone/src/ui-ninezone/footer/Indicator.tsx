/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
/** @packageDocumentation
 * @module Footer
 */

import "./Indicator.scss";
import classnames from "classnames";
import * as React from "react";
import { CommonProps } from "@bentley/ui-core";

/** Properties of [[FooterIndicator]] component.
 * @beta
 */
export interface FooterIndicatorProps extends CommonProps {
  /** Indicator content. */
  children?: React.ReactNode;
  /** Describes whether the footer is in footer or widget mode. */
  isInFooterMode?: boolean;
}

/** Indicator used in [[Footer]] component.
 * @beta
 */
export class FooterIndicator extends React.PureComponent<FooterIndicatorProps> {
  public render() {
    const className = classnames(
      "nz-footer-indicator",
      this.props.isInFooterMode && "nz-footer-mode",
      this.props.className);

    return (
      <div
        className={className}
        style={this.props.style}
      >
        {this.props.children}
      </div>
    );
  }
}
