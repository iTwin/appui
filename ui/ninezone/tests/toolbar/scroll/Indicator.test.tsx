/*---------------------------------------------------------------------------------------------
|  $Copyright: (c) 2018 Bentley Systems, Incorporated. All rights reserved. $
 *--------------------------------------------------------------------------------------------*/
import { mount, shallow } from "enzyme";
import * as React from "react";

import Indicator from "@src/toolbar/scroll/Indicator";

describe("<Indicator />", () => {
  it("should render", () => {
    mount(<Indicator />);
  });

  it("renders correctly", () => {
    shallow(<Indicator />).should.matchSnapshot();
  });
});
