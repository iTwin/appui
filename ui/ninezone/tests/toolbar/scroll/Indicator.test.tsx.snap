// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Indicator /> renders correctly 1`] = `
<div
  className="nz-toolbar-scroll-indicator nz-direction-left"
>
  <Chevron
    direction={0}
  />
</div>
`;
